### PR TITLE
dind: add demo IPsec cluster-wide encryption

### PR DESCRIPTION
**Pull request by dcbw**
All node<->master and node<->node communication is encrypted with
IPsec using cluster certificates for authentication.

@openshift/networking

### CODE REVIEW

**Comment by dcbw** [comment]
See also https://github.com/openshift/openshift-docs/pull/3236

**Comment by dcbw** [comment]
The only icky part here is grabbing the latest libreswan RPM directly from my repo, that will go away once we have 3.19 in F24 repos eventually.  So I don't think we should merge quite yet, but I want to get comments on the approach.

File: images/dind/Dockerfile [hunk 1]
```diff
@@ -72,6 +78,8 @@ RUN systemctl enable docker.service
 RUN echo "DOCKER_STORAGE_OPTIONS=--storage-driver vfs" >\
  /etc/sysconfig/docker-storage
 
+COPY openshift-dind-lib.sh /usr/local/bin/
```
Comment (marun) [inline images/dind/Dockerfile]: Why is this moving from the node image?  My intent was to allow the base dind image to be used for things other than running an image-based openshift cluster (e.g. as an ansible target).

Comment (marun) [inline images/dind/Dockerfile]: I suggest moving the libssl additions to the node image instead.

Comment (dcbw) [inline images/dind/Dockerfile]: @marun the master needs it too, which is why it's now shared in the base image between master and node.  Do you mean just have both master and node copy the file in?  I wasn't sure how to do that if the file wasn't actually in the current directory with a Dockerfile.  Can I do "COPY ../foobar"?

Comment (marun) [inline images/dind/Dockerfile]: The dependency graph is base -> node -> master, so the master gets anything on the node image.

**Comment by dcbw** [comment]
@marun updated, PTAL thanks!

**Comment by danwinship** [comment]
> The only icky part here is grabbing the latest libreswan RPM directly from my repo, that will go away once we have 3.19 in F24 repos eventually.

Did that happen?

> So I don't think we should merge quite yet, but I want to get comments on the approach.

It seems like something that really isn't going to get used very much, so maybe adding a bunch of code to the dind script for it isn't the best approach? Could this be done as a separate helper script? Or maybe we should get support for ipsec added to openshift-ansible and then when #11334 merges people could test ipsec-in-dind that way...

**Comment by smarterclayton** [comment]
Fair warning - when `openshift start` goes away dind is going to be a lot harder to maintain.  At that point I'd recommend just doing the work in ansible, because any other place is going to be a lot more complicated.

**Pull request closed without merging**